### DISCUSSION TITLE
Significant increase in download size as of 11.0.0-preview7 when targeting web-assembly compared to Playground

### DISCUSSION THREAD

**Discussion by koustubhmoharir** [AvaloniaUI/Avalonia#11277] [Q&A] [answered]
I started with the avalonia.xplat template and made the following changes to improve speed and reduce size. However, the total build size is significantly larger than the XamlPlayground although there are fewer actual dependencies. 
dotnet.wasm is larger, and it looks like all dlls that are a part of Microsoft.NETCore.App are being downloaded.

I also tried publishing with `dotnet publish -c Release`. It does not reduce the size but a lot of dlls seem to get bundled in dotnet.wasm.

I wanted to know if this is a genuine unavoidable increase in dependency size or a regression that will get fixed in future.

To reduce size, I have done the following:
Removed dependencies on ReactiveUI, Inter font, and switched from Fluent theme to Simple theme.
Added the following section to the project file by looking at the XamlPlayground
```
  <PropertyGroup Condition="'$(Configuration)'=='Release'">
    <RunAOTCompilation>true</RunAOTCompilation>
    <PublishTrimmed>true</PublishTrimmed>
    <TrimMode>full</TrimMode>
    <WasmBuildNative>true</WasmBuildNative>
    <InvariantGlobalization>true</InvariantGlobalization>
    <EmccCompileOptimizationFlag>-O3</EmccCompileOptimizationFlag>
    <EmccLinkOptimizationFlag>-O3</EmccLinkOptimizationFlag>
  </PropertyGroup>
```

**Comment by maxkatz6** [comment, marked as answer]
Have you built XamlPlayground on your machine to compare?
Keep in mind, web version is hosted on static websites host which supports compression. Modern browsers actually show you compressed size and unconfessed as well. 

> I wanted to know if this is a genuine unavoidable increase in dependency size or a regression that will get fixed in future.

In general, you need to follow Microsoft recommendations how to reduce app size. Trimming is one of the options.
Dependencies like reactive ui or additional fonts are quite fat as well. 

Note, you are using AOT on WASM, which heavily increases app size as well. With a little of perf benefits. 

In general, I don't think there were any big size regressions on Avalonia side, as I do check and compare it once in a while. Unless it was regressed somewhere in a last month.

**Reply by koustubhmoharir** [reply]
Thanks for the quick response. I must have copied the RunAOTCompilation property from some other sample repo, not the Playground. Turning it off reduces the size by more than half, and it is now better than I expected. The uncompressed size went down from 34.1 MB to 13.7 MB, compared to the 27 MB for the Playground.
